astropy/config/paths.py

```diff
@@ -167,20 +168,25 @@ def _get_dir_path(cls, rootname: str) -> Path:
                 path.mkdir(exist_ok=True)
             return path.resolve()
 
-        if (
-            (dir_ := os.getenv(cls._directory_env_var)) is not None
-            and (xch := Path(dir_)).exists()
-            and not (xchpth := xch / rootname).is_symlink()
-        ):
-            if xchpth.exists():
-                return xchpth.resolve()
+        if (env_dir_str := os.getenv(cls._directory_env_var)) is None:
+            return cls._find_or_create_root_dir(linkto=None, pkgname=rootname)
 
-            # symlink will be set to this if the directory is created
-            linkto = xchpth
-        else:
-            linkto = None
+        if not (env_dir_path := Path(env_dir_str)).is_dir():
+            warn(
+                f"{cls._directory_env_var} is set to {env_dir_str!r}, "
+                "but no such directory was found. "
+                "This environment variable will be ignored.",
+                category=UserWarning,
+                stacklevel=3,
+            )
+            return cls._find_or_create_root_dir(linkto=None, pkgname=rootname)
 
-        return cls._find_or_create_root_dir(linkto, rootname)
+        path = env_dir_path / rootname
+        path.mkdir(exist_ok=True)
+        if not path.is_symlink():
+            return path.resolve()
+        else:
+            return cls._find_or_create_root_dir(linkto=path, pkgname=rootname)
```

Comment:
I think this is the `main` behavior. (Still thinking through it...)
```suggestion
        if not path.is_symlink():
            if path.exists():
                return path.resolve()
            linkto = path
        else:
            linkto = None
        return cls._find_or_create_root_dir(linkto=path, pkgname=rootname)
```

Reply:
See my reasoning at https://github.com/astropy/astropy/pull/17934#pullrequestreview-3362248085